Source/WebCore/rendering/RenderLayerFilters.cpp

```diff
@@ -187,10 +198,15 @@ GraphicsContext* RenderLayerFilters::beginFilterEffect(RenderElement& renderer,
     }
 
     resetDirtySourceRect();
-    filter.setFilterRegion(m_filterRegion);
 
-    if (!m_targetSwitcher || hasUpdatedBackingStore)
-        m_targetSwitcher = FilterTargetSwitcher::create(context, filter, m_targetBoundingBox, DestinationColorSpace::SRGB());
+    if (!m_targetSwitcher || hasUpdatedBackingStore) {
+        FloatRect sourceImageRect;
+        if (auto* shape = dynamicDowncast<RenderSVGShape>(renderer))
+            sourceImageRect = shape->strokeBoundingBox();
```

Comment:
Is it sufficient to use `strokeBoundingBox()` here, not taking other visual effects into account?
You can directly access that method from `renderer`, no need to cast to `RenderSVGShape`, a simple `is<>` check is sufficient.

Reply:
Yes `sourceImageRect` is the rectangle that is needed to draw the `sourceImage`. While applying the filter, every `FilterEffect` adds the necessary outsets to its result `FilterImage`. The `filterRegion` should include the outsets.